packages/client/src/containers/CollectionsPage/CollectionsPage.Container.js

```diff
@@ -91,6 +98,34 @@ export const CollectionsPage = () => {
     setFilteredProducts(filterResults);
   }, [searchInput, products]);
 
+  const toggleFavorite = (id, title, price, event) => {
+    const newFavoriteItem = {
+      id,
+      title,
+      price,
+    };
+
+    const inFavoriteList = favoriteProducts.some((e) => {
+      return e.id === id;
+    });
+
+    if (!inFavoriteList) {
```

Comment:
I think this would be better if you used the `if` block to only decide about adding or removing the favorite and saving either path into a variable called `newFavorites`, then you'd only have one call to `setFavoriteProducts`and `setLocalStorage` (which again, could be hidden as a single custom hook).